common-lib/src/main/scala/com/gu/mediaservice/lib/metadata/SoftDeletedMetadataTable.scala

```diff
@@ -1,46 +1,46 @@
 package com.gu.mediaservice.lib.metadata
 
 import com.amazonaws.services.dynamodbv2.model.BatchWriteItemResult
-import com.gu.mediaservice.lib.aws.DynamoDB
 import com.gu.mediaservice.lib.config.CommonConfig
 import com.gu.mediaservice.model.ImageStatusRecord
-import com.gu.scanamo._
-import com.gu.scanamo.syntax._
+import org.scanamo._
+import org.scanamo.syntax._
+import org.scanamo.generic.auto._
+import software.amazon.awssdk.services.dynamodb.DynamoDbAsyncClient
 
 import scala.concurrent.{ExecutionContext, Future}
 import scala.jdk.CollectionConverters.collectionAsScalaIterableConverter
 
-class SoftDeletedMetadataTable(config: CommonConfig) extends DynamoDB[ImageStatusRecord](config, config.softDeletedMetadataTable) {
-  private val softDeletedMetadataTable = Table[ImageStatusRecord](table.getTableName)
+class SoftDeletedMetadataTable(config: CommonConfig) {
+  val client = config.withAWSCredentialsV2(DynamoDbAsyncClient.builder()).build()
+
+  private val softDeletedMetadataTable = Table[ImageStatusRecord](config.softDeletedMetadataTable)
 
   def getStatus(imageId: String)(implicit ex: ExecutionContext) = {
-    ScanamoAsync.exec(client)(softDeletedMetadataTable.get('id -> imageId))
+    ScanamoAsync(client).exec(softDeletedMetadataTable.get("id" === imageId))
   }
 
   def setStatus(imageStatus: ImageStatusRecord)(implicit ex: ExecutionContext) = {
-    ScanamoAsync.exec(client)(softDeletedMetadataTable.put(imageStatus))
+    ScanamoAsync(client).exec(softDeletedMetadataTable.put(imageStatus))
   }
 
-  private def extractUnprocessedIds(results: List[BatchWriteItemResult]): List[String] =
-    results.flatMap(_.getUnprocessedItems.values().asScala.flatMap(_.asScala.map(_.getPutRequest.getItem.get("id").getS)))
-
```

Comment:
scanamo no longer returns the list of unprocessed items; it looks like it internally will repeat until all items have been processed, so we should no longer have to care about it in our application code?! 
https://github.com/scanamo/scanamo/blob/3a68697525c8dde1e1aee757d2d1f729511106b7/scanamo/src/main/scala/org/scanamo/ScanamoFree.scala#L56-L64